embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/CurrentSessionSpanImpl.kt

```diff
@@ -55,52 +42,23 @@ internal class CurrentSessionSpanImpl(
      * be counted as such towards the limits, so make sure there's no case afterwards where a Span is not created.
      */
     override fun canStartNewSpan(parent: EmbraceSpan?, internal: Boolean): Boolean {
+        // Check conditions where no span can be created
         val currentSession = sessionSpan.get() ?: return false
         if (!currentSession.isRecording || (parent != null && parent.spanId == null)) {
             return false
         }
 
-        if (internal) {
+        // If a span can be created, always let internal spans be to be created
```

Comment:
It seems sensible to add some very high limit (e.g. 10_000)

Reply:
We should. I would be an SDK error if we go beyond that. Let me add a task to track this.

The good thing is that spans are pretty small if you don't put lots of events and attributes on them. 10K internal spans is probably less in size than 100 thick spans with lots of long attributes and events. I'll do some profiling and pick a reasonable value.